### ISSUE TITLE
[Bug]: Forced 360p

### ISSUE THREAD

**Issue by Bostonroller** [FreeTubeApp/FreeTube#8509]
### Guidelines

- [x] I have encountered this bug in the [latest release of FreeTube](https://github.com/FreeTubeApp/FreeTube/releases).
- [x] I have encountered this bug in the [official downloads of FreeTube](https://github.com/FreeTubeApp/FreeTube#official-downloads).
- [x] I have [searched the issue tracker for open and closed issues](https://github.com/FreeTubeApp/FreeTube/issues?q=is%3Aissue+sort%3Arelevance-desc) that are similar to the bug report I want to file, without success.
- [x] I have searched the [documentation](https://docs.freetubeapp.io/) for information that matches the description of the bug I want to file, without success.
- [x] This issue contains only one bug.
- [x] I have read and agree to follow the [rules](https://docs.freetubeapp.io/community/rules/).

### Describe the bug

Im trying to watch video 
https://youtu.be/GobLdUEccVY
But it keep getting forced to 360p or it is only audio and no video.
Local API is in use. Same problem was in earlier video in that channel, but it started to work when I changed resolution to 720p, but in this video it won't let to change the resolution.
Even if video is paused, it change resolution to 360p and start to play it automatically.

### Expected Behavior

Video should play normally 1080p.

### Issue Labels

accessibility issue, content not loading

### FreeTube Version

v0.23.12 Beta

### Operating System Version

Windows 11

### Installation Method

.exe

### Primary API used

Local API

### Additional Information

Other videos, atleast in other channels, work like they should.

### Nightly Build

- [ ] I have encountered this bug in the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds).

**Comment by Bostonroller** [comment]
I kept watching that video 360p and after halftime it started to changing to audio only. If I reload it, I can watch couple second, but then it pause and lose video again.

Same problem in his next video too...
https://youtu.be/J0FntyF77d8

**Comment by tsasam** [comment]
Have the same problem but with a different video

**Comment by Ein-Tim** [comment]
This issue is already tracked in https://github.com/FreeTubeApp/FreeTube/issues/7119

**Comment by Bostonroller** [comment]
> This issue is already tracked in [#7119](https://github.com/FreeTubeApp/FreeTube/issues/7119)

Ok, I didn't noticed it was also for video, not only audio issue.
I tried to use mvp player, but it didn't have audio at all over half of the video.
It does work if I download whole video.

[EDIT the next day]
It has never done this, but now it does this in EVERY video I watch in that channel. External Player does help, but in some videos it lost audio so it is not a best solution. I have to same time play the original video to get audio and use mvp to see the video at better quality...

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@Bostonroller please try the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds) and report back if this resolves the issue for you

**Comment by Bostonroller** [comment]
> [@Bostonroller](https://github.com/Bostonroller) please try the latest [nightly build](https://docs.freetubeapp.io/development/nightly-builds) and report back if this resolves the issue for you

Yes, I can try that.
With the latest beta 0.23.13, I have had this problem only once and I use FreeTube multiple hours per day.
When I made the report it was so obvious, but now when I tried this older beta 0.23.12 it happened only once even when I played 5 videos at the same time. So I can't find the way to replicate the issue currently. 

[EDIT]
I played that same video I reported earlier and couple more and it did lost other resolutions except 360p once, but video was playing at background so I didn't saw what happened. v0.23.13-nightly-6941 Beta is my current build.
Issue don't occur so often what it did when I made report, but it still seems to do it.
I keep testing.

Couple times it reloaded whole video, but it did continue with right resolution.

0.23.13-nightly-6944
"Reloading player according to SABR request" or something like that. It occur almost every video. And it do it multiple times in same video.
After over 10 videos, one is turned to 360p and it lost watching position.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@Bostonroller hmm i still cant reproduce this. Maybe this is location related. Would it be possible for you to provide from what country you are connecting from. Feel free to ask my email if you dont feel comfortable sending your location here.

Also are you using a VPN by any chance?

**Comment by Bostonroller** [comment]
Location is Finland. I don't use VPN. 
Situation is now much better than it was when I made report. I couldn't put videos to pause, because so often it forced 360p and continued playing video. Now, it won't do it so Im happy.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> Now, it won't do it so Im happy.

Just to be clear the forced 360p issue still persists for you on the nightly build but its just way less?

If that is the case could you try the following. Go to a video in FreeTube where you get a forced 360p. Now open your browser in a private window and navigate to the same video. Make sure you are not logged into YouTube or a Google account in the private window. 

Please share what YouTube displays for you

**Comment by Bostonroller** [comment]
It does persist. Got it one time today, same as yesterday. Most often it just reload whole video and select default resolution. 
When issue occurred I reloaded the video and it never turned to 360p again. 
The original link I posted in my report and couple next video on the channel, reloading didn't help at all. It keep turning to 360p. So currently nightly build is better.

I don't think Youtube display anything abnormal because it seems to be more random event than just specific video problem.  I can try that next time it occur.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@Bostonroller thanks for providing the information

In additional to going to YouTube to verify what happens on there i would like you to provide additional info while testing the nightly build


1. ctrl + shift + i on the subscription page to open the dev tools. Make sure the console tab in the dev tools is open

2. Go to the about page to show what build version you are on

3. Try to watch the video and reproduce the bug

4. Show the red errors you received in the console when the bug occured

Extra points for a screenrecording of these steps :)

**Comment by Bostonroller** [comment]
freetube-0.23.13-nightly-6949
This is screenshot when it do that reloading.
Ill try to catch 360p too.
First screenshot was when it reloaded it first time, then next two is after couple minutes when it did it again. After reloading there can be 2-3 SABR backoff timer at the same time.

![Image](https://github.com/user-attachments/assets/ce77eb1f-a2f2-46a9-8d70-c43d2f7fe6fa)
![Image](https://github.com/user-attachments/assets/7a491d2b-93c9-45eb-81ec-f863d9731705)
![Image](https://github.com/user-attachments/assets/dfcb37d2-af61-4060-b41b-dbe9cb60ccc8)

Here is one more reloading event

![Image](https://github.com/user-attachments/assets/c95220d8-ed8b-488a-ba02-6f4589d61b9b)

**Comment by Bostonroller** [comment]
Ok, here is console error from 360p incident. 

![Image](https://github.com/user-attachments/assets/093a36d0-065f-4870-98d2-83f0a8d97cd6)
![Image](https://github.com/user-attachments/assets/02655849-36dd-4840-8469-997c31e2ffff)
 Another one
![Image](https://github.com/user-attachments/assets/55d3d864-ccbc-4eb1-9ee2-fa1bebdf198a)

I was playing video in private windows at the same time, but it didn't have any problems.
Youtube just spam this same message in console while playing it.

![Image](https://github.com/user-attachments/assets/c5886961-edb5-4e4b-99b1-a6df7d3bfa32)

Yet another 360p error

![Image](https://github.com/user-attachments/assets/86914b53-e5a3-4aca-92cb-2259599d7680)

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> [@Bostonroller](https://github.com/Bostonroller) thanks for providing the information
> 
> In additional to going to YouTube to verify what happens on there i would like you to provide additional info while testing the nightly build
> 
>     1. ctrl + shift + i on the subscription page to open the dev tools. Make sure the console tab in the dev tools is open
> 
>     2. Go to the about page to show what build version you are on
> 
>     3. Try to watch the video and reproduce the bug
> 
>     4. Show the red errors you received in the console when the bug occured
> 
> 
> Extra points for a screenrecording of these steps :)

The current output is a bit strange. Could you try these steps with the nightly build again but make a video. Upload limit on GitHub is 10MB

**Comment by Bostonroller** [comment]
> The current output is a bit strange. Could you try these steps with the nightly build again but make a video. Upload limit on GitHub is 10MB

Ill try, but it seems to be really hard. I recorded dozens videos, but when I record, nothing odd happens. But those couple times when I didn't record, I got the problem...
Still I managed to get one record of the reloading problem. That usually happen really often. 
Here is that clip. Dev tools was not open, but they only say what I posted earlier.
Ill try to capture that 360p too if it's necessary.

https://github.com/user-attachments/assets/37ae9ef1-8460-4407-b758-893202b4d616

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
@Bostonroller im a bit confused what this clip is trying to show me.

**Comment by Bostonroller** [comment]
> [@Bostonroller](https://github.com/Bostonroller) im a bit confused what this clip is trying to show me.

That is biggest problem currently. It keep reloading the video multiple times per video. And when it does that, it change resolution and captions to default and if you want to use captions and different resolution on video you are watching, it could be 5-10 second and it reload it again and you have to change those setting again.
It does that often. Earlier there was no such problem.
It just say "Reloading player according to SABR request"

Other problem is when video just flash and there is no "Reloading player according to SABR request" or anything, but video is stuck to 360p and there are no other resolutions to select. 
THEN you have to reload it to make it work again.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> That is biggest problem currently. It keep reloading the video multiple times per video. And when it does that, it change resolution and captions to default and if you want to use captions and different resolution on video you are watching, it could be 5-10 second and it reload it again and you have to change those setting again.
It does that often. Earlier there was no such problem.
It just say "Reloading player according to SABR request"

Okay i understand now but isnt related to the forced 360p issue so lets focus on that specific first


> And when it does that, it change resolution and captions to default

I would like to zoom into this more though. What do you exactly mean with this?



> Ill try to capture that 360p too if it's necessary.

Please do because that is what this issue is about.

**Comment by Bostonroller** [comment]
> I would like to zoom into this more though. What do you exactly mean with this?


It means those settings you set from Settings-Player, Enable Subtitles by Default and Default Quality.
I use subtitles only if I watch videos from Asia because they don't speak English that often and I also like to use HD quality and change it higher if necessary.
If you reload the player, it use those setting you have set. So it is kind of annoying to enable subtitle or change resolution if it reload the player multiple times.

This issue with 360p seems to be really rare now. It happen only once per day or even less so I can't make a video of it. It seems that issue is resolved or atleast it's not a huge problem any longer.

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
> This issue with 360p seems to be really rare now. It happen only once per day or even less so I can't make a video of it. It seems that issue is resolved or atleast it's not a huge problem any longer.

We would like to solve this before it reaches others so please screen capture it with the console open so we can see the logs in real time.

> It means those settings you set from Settings-Player, Enable Subtitles by Default and Default Quality.
I use subtitles only if I watch videos from Asia because they don't speak English that often and I also like to use HD quality and change it higher if necessary.
If you reload the player, it use those setting you have set. So it is kind of annoying to enable subtitle or change resolution if it reload the player multiple times.

This is interesting. I would like to see a screenrecording of this behavior too

**Comment by Bostonroller** [comment]
Here it is with console open. 

https://github.com/user-attachments/assets/5d34c5e7-e9b2-423d-b3f9-779e640aa896

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
Hi @Bostonroller sorry for the delay. Is this still happening for you in v0.24.0?

**Comment by Bostonroller** [comment]
> Hi [@Bostonroller](https://github.com/Bostonroller) sorry for the delay. Is this still happening for you in v0.24.0?

Yes, I understand why there was a delay.
It seems to still happening...

<img width="1215" height="1139" alt="Image" src="https://github.com/user-attachments/assets/7aa24998-36c8-4221-b7cc-4d668cc3704d" />

**Comment by Bostonroller** [comment]
I got this new error message just before it went to 360p.
Never had it before.
"There are no comments available for this video
Local API Error (Click to copy): TypeError: Cannot read properties of undefined (reading 'metadata_parts')"

<img width="283" height="1271" alt="Image" src="https://github.com/user-attachments/assets/837c8a67-af9b-4b59-a7ac-79a3999fc000" />

<img width="259" height="1053" alt="Image" src="https://github.com/user-attachments/assets/f515dd8e-e167-41df-9e74-f41264def3da" />

**Comment by hyperland956-lgtm** [comment]
Hello, I'm not entirely sure how my issue (#9459) is related to this one, as the bug isn't about the resolution dropping to 360p, so I’m providing additional context and a screen rec here to show exactly what’s happening in the latest beta.

While this specific thread here focuses on the player forcing 360p, my issue is specifically regarding **codec selection behavior**... The video shows that even with DASH enabled, FreeTube is consistently always defaulting to the legacy `avc1` codec across multiple different videos, rather than using vp9 or av1. The audio stream selection is also inconsistent, switching between mp4a and opus.

Here is a screen rec showing this behavior across a variety of videos: 

https://github.com/user-attachments/assets/25417751-2c41-4afb-a26f-efef4090046f



Environment:
* OS: Windows 10
* FreeTube Version: v0.25.0-beta
* Region: Asia (I tried with US vpn but still not resolved)
* Default Player Type: DASH

**Comment by Bostonroller** [comment]
For me this case is solved. It has been long time since I had this problem last time.  FreeTube seems to work really well now.
